src/libraries/System.Private.CoreLib/src/System/Diagnostics/Debug.cs

```diff
@@ -389,10 +396,14 @@ public WriteIfInterpolatedStringHandler(int literalLength, int formattedCount, b
             }
 
             /// <summary>Extracts the built string from the handler.</summary>
-            internal new string ToString() =>
-                _stringBuilderHandler._stringBuilder is StringBuilder sb ?
-                    sb.ToString() :
+            internal string ToStringAndClear()
+            {
+                string s = _stringBuilderHandler._stringBuilder is StringBuilder sb ?
+                    StringBuilderCache.GetStringAndRelease(sb) :
```

Comment:
Remind me - we normally expect this `StringBuilder` instance to be released unless something goes wrong during the format operation itself, correct? (Because if something fails during the format operation then _ToStringAndClear_ never gets called?)

I think the behavior here is sensible, just wanted to confirm my understanding. :)

Reply:
> we normally expect this StringBuilder instance to be released unless something goes wrong during the format operation itself, correct?

Correct